RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -500,9 +717,134 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
 	      }
           } // end of HBHE check
      } // loop on HcalCalibDigiCollection
+
+     // now match the laser monitor data by fiber (in time) 
+     if( fillLaserMonitor_ ) {
+       // check for any fibers without data and fill
+       // them so we dont run into problems later
+       for( unsigned idx = 0; idx < laserMonDetTypeList_.size(); ++idx ) {
+           if( lasmon_adcs[idx].empty() ) {
+               for( int i = 0; i < 10; ++i ) {
+                   lasmon_adcs[idx].push_back( -1 );
+               }
+           }
+           if( lasmon_capids[idx].empty() ) {
+               for( int i = 0; i < 10; ++i ) {
+                   lasmon_capids[idx].push_back( -1 );
+               }
+           }
+       }
+       int nFibers = laserMonIEtaList_.size();
+       // for each fiber we need to find the index at with the 
+       // data from the next fiber matches in order to stitch them together.
+       // When there is an overlap, the data from the end of the
+       // earlier fiber is removed.  There is no removal of the last fiber
+       std::map<int, int> matching_idx; 
```

Comment:
it looks like this can be a vector of ints

Reply:
As in the other maps, it seems that indeed the map is not necessary.  I've changed it